internal/character/qingque/card.go

```diff
@@ -0,0 +1,144 @@
+package qingque
+
+func (c *char) swap(pos1, pos2 int) {
+	c.tiles[pos1], c.tiles[pos2] = c.tiles[pos2], c.tiles[pos1]
+	c.suits[pos1], c.suits[pos2] = c.suits[pos2], c.suits[pos1]
+}
+
+// Functions if distinct suits aren't needed to save time and memory, for documentation read the comments on the functions including distinct suits
+func (c *char) drawTile() string {
+	c.e2()
+	if c.tiles[0] != 4 {
+		s1, s2, s3 := c.tiles[0], c.tiles[1], c.tiles[2]
+		startingTiles := s1 + s2 + s3
+		drawn := c.engine.Rand().Intn(3)
+		c.tiles[drawn] += 1
+		if c.tiles[0] >= c.tiles[1] && c.tiles[1] >= c.tiles[2] {
+		} else if c.tiles[1] > c.tiles[0] {
+			c.swap(0, 1)
+		} else if c.tiles[2] > c.tiles[0] {
+			c.swap(0, 2)
+		} else {
+			c.swap(1, 2)
+		}
```

Comment:
nit: use switch/case where having a blank case is fine (add comment explaining that blank on purpose)